src/modules/RemediationsModal/common/ExistingPlaybookTypeahead.js

```diff
@@ -0,0 +1,204 @@
+  formOptions,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [filterValue, setFilterValue] = useState('');
+  const [selectOptions, setSelectOptions] = useState(existingRemediations);
+  const [focusedItemIndex, setFocusedItemIndex] = useState(null);
+  const [activeItem, setActiveItem] = useState(null);
+  const textInputRef = useRef();
+
+  useEffect(() => {
+    let newSelectOptions = existingRemediations;
+    if (filterValue) {
+      newSelectOptions = existingRemediations.filter((menuItem) =>
+        String(menuItem.name).toLowerCase().includes(filterValue.toLowerCase())
+      );
+
+      if (!newSelectOptions.length) {
+        newSelectOptions = [
+          {
+            name: `No results found for "${filterValue}"`,
+          },
+        ];
+      }
+
+      if (!isOpen) {
+        setIsOpen(true);
+      }
+    }
+
+    setSelectOptions(newSelectOptions);
+    setActiveItem(null);
+    setFocusedItemIndex(null);
+  }, [filterValue]);
+
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const onSelect = (_event, value) => {
+    setIsLoadingRemediation(true);
+    api.getRemediation(value).then((remediation) => {
+      setSelectedPlaybook(remediation);
+      setIsLoadingRemediation(false);
+      existingPlaybookSelected && input.onChange(remediation.name);
+      existingPlaybookSelected &&
+        formOptions.change(EXISTING_PLAYBOOK, remediation);
+    });
+  };
+
+  const onTextInputChange = (_event, value) => {
+    setFilterValue(value);
+  };
+
+  const handleMenuArrowKeys = (key) => {
+    let indexToFocus;
+
+    if (isOpen) {
+      if (key === 'ArrowUp') {
+        if (focusedItemIndex === null || focusedItemIndex === 0) {
+          indexToFocus = selectOptions.length - 1;
+        } else {
+          indexToFocus = focusedItemIndex - 1;
+        }
+      }
+
+      if (key === 'ArrowDown') {
+        if (
+          focusedItemIndex === null ||
+          focusedItemIndex === selectOptions.length - 1
+        ) {
+          indexToFocus = 0;
+        } else {
+          indexToFocus = focusedItemIndex + 1;
+        }
+      }
+
+      setFocusedItemIndex(indexToFocus);
+      const focusedItem = selectOptions.filter((option) => !option.isDisabled)[
+        indexToFocus
+      ];
+
+      setActiveItem(`select-typeahead-${focusedItem.name.replace(' ', '-')}`);
+    }
+  };
+
+  const onInputKeyDown = (event) => {
+    const enabledMenuItems = selectOptions.filter(
+      (option) => !option.isDisabled
+    );
+    const [firstMenuItem] = enabledMenuItems;
+    const focusedItem = focusedItemIndex
+      ? enabledMenuItems[focusedItemIndex]
+      : firstMenuItem;
+    switch (event.key) {
+      case 'Enter':
+        if (isOpen && focusedItem.name !== 'no results') {
+          onSelect(null, focusedItem.id);
+        }
+        setIsOpen((prevIsOpen) => !prevIsOpen);
+        setFocusedItemIndex(null);
+        setActiveItem(null);
+        break;
+      case 'Escape':
```

Comment:
This is causing the wizard to close instead of the select dropdown. I believe this is unexpected, or?